### DISCUSSION TITLE
Titles generated by `summarizeText` frequently include quotation marks.

### DISCUSSION THREAD

**Discussion by CNSeniorious000** [anse-app/anse#18] [General]
As the title suggests.

I tried to prompt engineering to force it not to produce quotes but proved ineffective. With the goal of conserving tokens in mind, I think we can use regular expressions to remove the quotes from the title as a post-processing step.

In `src/logics/conversation.ts`:

```diff
  // Update conversation title
  if (providerResponse && bot.type === 'chat_continuous' && !conversation.name) {
    const inputText = conversation.systemInfo || prompt!
    const rapidPayload = generateRapidProviderPayload(promptHelper.summarizeText(inputText), provider.id)
    const generatedTitle = await getProviderResponse(provider.id, rapidPayload).catch(() => {}) as string || inputText
    updateConversationById(conversation.id, {
-     name: generatedTitle,
+     name: generatedTitle.replace(/^['"\s]+|['"\s]+$/g, ''),
    })
  }
```

It works fine! You can check it out at my free deploy https://anse.up.railway.app/ ✨

How do you think of it?

**Comment by ddiu8081** [comment]
I think this is a good change, pr welcome.

**Comment by CNSeniorious000** [comment]
I find a bug I made in #22. For example:

- `generatedTitle` is `What is "xxx"`
- `generatedTitle.replace(/^['"\s]+|['"\s]+$/g, '')` is `What is "xxx`

This is bad. So I recommend revert this commit or change it as follow:

```diff
    updateConversationById(conversation.id, {
-     name: generatedTitle.replace(/^['"\s]+|['"\s]+$/g, ''),
+     name: (generatedTitle.startsWith('"') && generatedTitle.endsWith('"')) ? generatedTitle.slice(1, -1) : generatedTitle
    })
```

Sorry about making a buggy PR.
